### PR TITLE
Added checking of running quote hits, to stop repeats running concurrently.

### PR DESCRIPTION
**Pull request by MitchPetersen**
Implements task dictionary and checking in LegacyStockServerHandler. Moves some functionality into LecacyStockServerHandler.

### CODE REVIEW

File: CacheService/docker_context/LegacyStockServerHandler.py [hunk 1]
```diff
@@ -17,6 +25,26 @@ async def quoteSocket(self):
...
+            check = await self.RedisHandler.rExists(stock_symbol)
+            if check:
+                quote = await self.RedisHandler.rGet(stock_symbol)
+                return quote['price'], 200
+            else:
+                return await self.newQuoteTask(trans_num, user_id, stock_symbol)
+        else:
+            return await self.newQuoteTask(trans_num, user_id, stock_symbol)
+
+    async def newQuoteTask(self, trans_num, user_id, stock_symbol):
```
Comment (MitchellGRead) [inline CacheService/docker_context/LegacyStockServerHandler.py]: should make this private to the class

File: CacheService/docker_context/LegacyStockServerHandler.py [hunk 2]
```diff
@@ -17,6 +25,26 @@ async def quoteSocket(self):
         return reader, writer
 
     async def getQuote(self, trans_num, user_id, stock_symbol):
+        if stock_symbol in self.tracker.keys():
+            await self.tracker[stock_symbol]
```
Comment (MitchellGRead) [inline CacheService/docker_context/LegacyStockServerHandler.py]: won't this just rerun the task? shouldn't we check to see if the task is done or not

**Review by MitchellGRead — COMMENTED** [review]
Just a couple of comments

File: CacheService/docker_context/LegacyStockServerHandler.py (hunk 2 above)
Comment (MitchPetersen) [reply CacheService/docker_context/LegacyStockServerHandler.py]: Old ways of thought die hard.

File: CacheService/docker_context/LegacyStockServerHandler.py [hunk 3]
```diff
@@ -17,6 +25,28 @@ async def quoteSocket(self):
         return reader, writer
 
     async def getQuote(self, trans_num, user_id, stock_symbol):
+        if stock_symbol in self.tracker.keys():
+            runningTask = self.tracker[stock_symbol]
+            while not runningTask.done():
+                sleep(1)
```
Comment (MitchellGRead) [inline CacheService/docker_context/LegacyStockServerHandler.py]: This is a blocking call, use `await asyncio.sleep`

File: CacheService/docker_context/LegacyStockServerHandler.py [hunk 4]
```diff
@@ -17,6 +25,28 @@ async def quoteSocket(self):
         return reader, writer
 
     async def getQuote(self, trans_num, user_id, stock_symbol):
+        if stock_symbol in self.tracker.keys():
+            runningTask = self.tracker[stock_symbol]
```
Comment (MitchellGRead) [inline CacheService/docker_context/LegacyStockServerHandler.py]: shouldn't it get removed when it does finish running

File: CacheService/docker_context/LegacyStockServerHandler.py (hunk 1 above)
Comment (MitchPetersen) [reply CacheService/docker_context/LegacyStockServerHandler.py]: done.

File: CacheService/docker_context/LegacyStockServerHandler.py (hunk 3 above)
Comment (MitchPetersen) [reply CacheService/docker_context/LegacyStockServerHandler.py]: done.

File: CacheService/docker_context/LegacyStockServerHandler.py (hunk 4 above)
Comment (MitchPetersen) [reply CacheService/docker_context/LegacyStockServerHandler.py]: self.tracker[stock_symbol] is removed at the end of __getQuoteFunc

**Review by MitchellGRead — APPROVED** [review]

**Pull request merged**